pkg/operator/workloadcontroller/workload_controller.go

```diff
@@ -276,3 +288,74 @@ func (c *ServiceCatalogAPIServerOperator) namespaceEventHandler() cache.Resource
 		},
 	}
 }
+
+func (c ServiceCatalogAPIServerOperator) deleteOwnerRefFromServiceBindings() error {
+	// Get service bindings
+	gvr := schema.GroupVersionResource{
+		Group:    "servicecatalog.k8s.io",
+		Version:  "v1beta1",
+		Resource: "servicebindings"} // lowercase plural is important
+
+	// get all of the service bindings
+	bindings, err := c.dynamicClient.Resource(gvr).List(metav1.ListOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Info("No service bindings found, nothing to delete.")
+			return nil
+		}
+		klog.Errorf("Problem getting the bindings list. %v", err)
+		return err
+	}
+
+	klog.Infof("we have %d binding items", len(bindings.Items))
+	for _, binding := range bindings.Items {
+		// get the secret name
+		secretName, exists, err := unstructured.NestedString(binding.Object, "spec", "secretName")
+		if err != nil {
+			klog.Errorf("binding %s has a secretName field that is not a string: %v", binding.GetName(), err)
```

Comment:
Nit: `NestedString()` also returns an error if it can't traverse `binding.Object` (in which case, `secretName` might not actually exist. Maybe make the error a little more generic?

```suggestion
			klog.Errorf("could not get secretName string from binding %s: %v", binding.GetName(), err)
```